core/src/test/java/org/apache/iceberg/view/TestViewMetadataParser.java

```diff
@@ -308,4 +322,57 @@ public void replaceViewMetadataWithMultipleSQLsForDialect() throws Exception {
 
     assertThat(replaced.currentVersion()).isEqualTo(viewVersion);
   }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"none", "gzip"})
+  public void metadataCompression(String codecName) throws IOException {
+    Codec codec = Codec.fromName(codecName);
+    String location = Paths.get(tmp.toString(), "v1" + getFileExtension(codec)).toString();
```

Comment:
I don't think that this should rely on `getFileExtension`. It's fine that it is used by `BaseViewOperations` to name the file, but there's no need to do that here. Instead, I think this should be parameterized by file name (`v1.metadata.json` and `v1.gz.metadata.json`) so that it is obvious what happens based on the file name in the test. Otherwise this is testing that the extension produced by `getFileExtension` triggers some behavior in the parser, rather than a specific extension triggers behavior in the parser.

Reply:
I was aligning this test with how compression was tested for table metadata in https://github.com/apache/iceberg/blob/c07f2aabc0a1d02f068ecf1514d2479c0fbdd3b0/core/src/test/java/org/apache/iceberg/TableMetadataParserTest.java#L63 but I've changed it to have explicit file names